src/base/datetime/utils/index.ts

```diff
@@ -38,7 +38,48 @@ const getDateObj = (date: Date) => {
   const attr = `${year}-${month}-${day}`;
   return { text, attr };
 };
+export const formatDateByLocale = (date: string, language: string = "ja") => {
+  const tempDate = new Date(date);
+  if (!isValidDate(tempDate)) {
+    return date;
+  }
+  const year = tempDate.getFullYear();
+  const month = padStart(tempDate.getMonth() + 1);
+  const day = padStart(tempDate.getDate());
+  if (language === "ja" || language === "zh") return `${year}-${month}-${day}`;
+  return `${month}/${day}/${year}`;
+};
 
+const isValidDate = (d: Date) => {
+  return d instanceof Date && !isNaN(d.getTime());
+};
+export const isValidDateFormat = (dateString: string, language: string) => {
```

Comment:
@h000780 @nmanhit 
You can using regex for simply validation:
```
const isValidDateFormat = (dateString, locale) => {
   const enRegex = /^(0[1-9]|1[0-2])\/(0[1-9]|[12][0-9]|3[01])\/\d{4}$/;
   if (locale === "en") {
     return dateString.match(enRegex) !== null;
   }

   const jaRegex = /^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/g;
   return dateString.match(jaRegex) !== null;
}

console.log(isValidDateFormat("2021-15-01", "ja"));

console.log(isValidDateFormat("2021-11-01", "ja"));

console.log(isValidDateFormat("15/01/2021", "en"));

console.log(isValidDateFormat("11/01/2021", "en"));
```

Reply:
Hi.
This regex rule cannot determine the February 29th of a leap year and some invalid dates, such as 2021-04-31 2020-2-29.

Reply:
@h000780 
How about check valid date before regex checking:
```
const isValidDateFormat = (dateString, locale) => {
  const dateObj = new Date(dateString);
  const notExistedDate = dateObj.getDate() !== parseInt(dateString.split("-")[2]);
  if (notExistedDate) return false;

  const enRegex = /^(0[1-9]|1[0-2])\/(0[1-9]|[12][0-9]|3[01])\/\d{4}$/;
  if (locale === "en") {
    return dateString.match(enRegex) !== null;
  }
  
  const jaRegex = /^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/g;
  return dateString.match(jaRegex) !== null;
}
```